examples/gallery/main.cpp

```diff
@@ -5,5 +5,65 @@
 
 int main()
 {
-    App::create()->run();
+    auto demo = App::create();
+
+    auto row_data = std::make_shared<slint::VectorModel<
+            std::shared_ptr<slint::Model<slint::private_api::StandardListViewItem>>>>();
+
+    for (int r = 1; r < 101; r++) {
+
+        auto items =
+                std::make_shared<slint::VectorModel<slint::private_api::StandardListViewItem>>();
+
+        for (int c = 1; c < 5; c++) {
+            slint::SharedString text("item");
+            text = text + slint::SharedString::from_number(c) + slint::SharedString(".")
+                    + slint::SharedString::from_number(r);
+            items->push_back(slint::private_api::StandardListViewItem { text, c == 1 });
+        }
+
+        row_data->push_back(items);
+    }
+
+    demo->global<TableViewPageAdapter>().set_row_data(row_data);
+
+    demo->global<TableViewPageAdapter>().on_sort_ascending(
+            [row_data, demo = slint::ComponentWeakHandle(demo)](int index) {
+                auto demo_lock = demo.lock();
+                (*demo_lock)->global<TableViewPageAdapter>().set_row_data(row_data);
+
+                (*demo_lock)
+                        ->global<TableViewPageAdapter>()
+                        .set_row_data(
+                                std::make_shared<slint::SortModel<std::shared_ptr<
+                                        slint::Model<slint::private_api::StandardListViewItem>>>>(
+                                        (*demo_lock)->global<TableViewPageAdapter>().get_row_data(),
```

Comment:
Here's an idea for a simplification. Perhaps this would compile?

```suggestion
                (*demo_lock)
                        ->global<TableViewPageAdapter>()
                        .set_row_data(
                                std::make_shared<slint::SortModel<std::shared_ptr<
                                        slint::Model<slint::private_api::StandardListViewItem>>>>(
                                        row_data,
```